### DISCUSSION TITLE
Understanding spiking behavior in Izhikevich nodes and voltage injection

### DISCUSSION THREAD

**Discussion by peschn** [BindsNET/bindsnet#528] [Q&A] [answered]
Hi,
first of all, thanks a lot for this fantastic project.

I got a question around the implementation of the different Nodes (neuron models). In most Nodes (except Izhikevich), state variables like voltage are first updated before spiking is checked. In Izhikevich, it is the other way around, i.e. it is first checked, whether the neurons spikes or not, before state variables are updated. What is the reason for this? To my understanding, it always requires two timesteps in network.run to identify spiking for Izhikevich nodes, i.e. first timestep updates state variables, second timestep checks spiking behavior.

Secondly, when looking at Network.run() it seems like that layers are first updated via layer.forward(...), before voltage is injected to the neurons via layer.v += inject_v. This implies that it requires two timesteps to detect spiking behavior. What is the reason for this behavior? 

Thanks a lot and best,

Peter

**Comment by Hananel-Hazan** [comment, marked as answer]
Good catch, Izhikevich neuron should behave in a similar stages like other neurons.  Same goes for the Vmem injection. 
I open a pull request #529 with and I will merge the changes.

**Reply by peschn** [reply]
Thanks a lot for the fast reply and the fixes!
